lua/apg/modules/ghosting.lua

```diff
@@ -316,28 +355,29 @@ local function checkDoor(ply, ent)
 				v:SetVelocity(push)
 			end
 		end
-		
+
 		timer.Simple(1, function()
 			if IsValid(ply) and IsValid(ent) then
 				ent.APG_Ghosted = false
 				ent:oldFadeDeactivate()
 				ent:SetCollisionGroup( COLLISION_GROUP_INTERACTIVE )
 
 				if IsValid(isTrap) then
-					APG.userNotification( "Unable to unstuck objects from fading door!", ply, 1 )
+					APG.log("CantUnstuck " .. ply:GetName() .. " from " .. ent)
+					APG.notification( "Unable to unstuck objects from fading door!", ply, 1 )
 					APG.entGhost(ent)
 				end
 			end
 		end)
 	end
 end
 
-APG.hookRegister(mod, "APG.FadingDoorToggle", "APG_FadingDoor", function(ent, isFading)
+APG.hookAdd(mod, "APG.FadingDoorToggle", "APG_FadingDoor", function(ent, isFading)
 	if APG.isBadEnt(ent) and APG.cfg["fadingDoorGhosting"].value then
 		local ply = APG.getOwner( ent )
 
-		if (IsValid(ply) and not isFading) then
-			timer.Simple(0.001, function() 
+		if (ply:IsPlayer() and not isFading) then

```

Comment:
Couldn't this err if `ply` is `nil`?

Perhaps:

```suggestion
local isValidPlayer = IsValid(ply) and ply:IsPlayer()
local isNotFading = not isFading
if isValidPlayer and isNotFading then
```

Reply:
Nice catch, yes it would error, but I don't think we should stick to original formatting.

Reply:
👍